libraries/rush-lib/src/logic/operations/ShellOperationRunnerPlugin.ts

```diff
@@ -40,13 +40,21 @@ function createShellOperations(
     if (phase && project && !operation.runner) {
       // This is a shell command. In the future, may consider having a property on the initial operation
       // to specify a runner type requested in rush-project.json
-      const customParameterValues: ReadonlyArray<string> = getCustomParameterValuesForPhase(phase);
+      let customParameterValues: ReadonlyArray<string> = getCustomParameterValuesForPhase(phase);
+
+      if (operation.shard) {
+        customParameterValues = [
+          ...customParameterValues,
+          `--shard=${operation.shard.current}/${operation.shard.total}`
+        ];
+      }
```

Comment:
Instantiate your own `ShellOperationRunner` instances inside of `ShardedPhaseOperationPlugin` for sharded operations. The `ShellOperationRunnerPlugin` has no business knowing anything about sharding.

Reply:
Updated in [7d34e74](https://github.com/microsoft/rushstack/pull/4652/commits/7d34e74bc94de34fb1bafe0c89cde48b7d01c7de).